packages/react-native/scripts/cocoapods/utils.rb

```diff
@@ -195,16 +195,23 @@ def self.apply_xcode_15_patch(installer, xcodebuild_manager: Xcodebuild)
     private
 
     def self.add_build_settings_to_pod(installer, settings_name, settings_value, target_pod_name, configuration_type)
+        # [macOS
+        valid_suffixes = [""]
+        if $RN_PLATFORMS != nil && $RN_PLATFORMS.length() >= 2
+            # Only apply suffixes if there are at least two platforms
+            valid_suffixes = $RN_PLATFORMS.map { |platform| "-#{platform}" }
+        end
+        # macOS]
         installer.target_installation_results.pod_target_installation_results.each do |pod_name, target_installation_result|
-            if pod_name.to_s == target_pod_name
+            if valid_suffixes.any? { |suffix| pod_name.to_s == "#{target_pod_name}#{suffix}" } # [macOS]
                 target_installation_result.native_target.build_configurations.each do |config|
-                        if configuration_type == nil || (configuration_type != nil && config.type == configuration_type)
-                            config.build_settings[settings_name] ||= '$(inherited) '
-                            config.build_settings[settings_name] << settings_value
-                        end
+                    if configuration_type == nil || (configuration_type != nil && config.type == configuration_type)
+                        config.build_settings[settings_name] ||= '$(inherited) '
+                        config.build_settings[settings_name] << settings_value
```

Comment:
Should we make an upstream fix so this logic matches what is done in `create_header_search_path_for_frameworks` in RN Core?

Reply:
Seems like a good idea. Let's revisit once this PR makes it in.